### PR TITLE
Include PRODUCT_SUFFIX in J9PRODUCT_NAME

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/16449

Tested via https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/312/ along with https://github.com/pshipton/openj9/tree/prodsuffix

With --with-product-suffix="OpenPete Edition":

-Xinternalversion `Eclipse OpenJ9 OpenJDK OpenPete Edition 64-bit Server VM (11.0.18-internal+0-adhoc.jenkins.BuildJDK11aarch64macPersonal) from macosx-aarch64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 11.0.18.0-ea, built on Dec 13 2022 ...` (removed the trailing build id and compiler info for clarity)

javacore contains `1CIPRODUCT     OpenJDK OpenPete Edition`

---

Also https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/313/ that doesn't set a product suffix.
However there is a default set in https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/make/autoconf/version-numbers#L44

-Xinternalversion `Eclipse OpenJ9 OpenJDK Runtime Environment 64-bit Server VM (11.0.18-internal+0-adhoc.jenkins.BuildJDK11aarch64macPersonal) from macosx-aarch64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 11.0.18.0-ea, built on Dec 13 2022 ...`

javacore contains `1CIPRODUCT     OpenJDK Runtime Environment`

---

Previous behaviour:

-Xinternalversion `Eclipse OpenJ9 OpenJDK 64-bit Server VM (11.0.18-internal+0-adhoc.jenkins.BuildJDK11aarch64macNightly) from macosx-aarch64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 11.0.18.0-ea, built on Dec 10 2022 ...`

javacore contains `1CIPRODUCT     OpenJDK`

### CODE REVIEW

**Comment by pshipton** [comment]
I'll work on the other versions after this is reviewed.

File: closed/openj9_version_info.h.in [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #define OPENJ9_VERSION_INFO_H
 
 #define J9COMPILER_VERSION_STRING "@COMPILER_VERSION_STRING@"
-#define J9PRODUCT_NAME            "@PRODUCT_NAME@"
+#define J9PRODUCT_NAME            "@PRODUCT_NAME@ @PRODUCT_SUFFIX@"
```
Comment (keithc-ca) [inline closed/openj9_version_info.h.in:28]: If `PRODUCT_SUFFIX` is empty, I don't think we want the embedded space.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/openj9_version_info.h.in (hunk 1 above)
Comment (pshipton) [reply closed/openj9_version_info.h.in]: I switched it to use RUNTIME_NAME, but turns out there is a default PRODUCT_SUFFIX.
https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/make/autoconf/version-numbers#L44

Comment (keithc-ca) [reply closed/openj9_version_info.h.in]: We don't need to worry about an embedded space. If you configure with `--with-product-suffix=""` it works as if you said `--with-product-suffix="Runtime Environment"`. So your earlier macro should work:
```
#define J9PRODUCT_NAME            "@PRODUCT_NAME@ @PRODUCT_SUFFIX@"
```

Comment (pshipton) [reply closed/openj9_version_info.h.in:28]: Updated.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins compile amac jdk11

**Pull request merged**